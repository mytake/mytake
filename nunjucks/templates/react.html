{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
