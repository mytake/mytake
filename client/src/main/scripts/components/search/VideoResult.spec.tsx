/*
 * MyTake.org website and tooling.
 * Copyright (C) 2018-2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import VideoResult from "./VideoResult";
import { kennedyNixon } from "../../utils/testUtils";

test("VideoResultPreview containing", () => {
  const tree = renderer
    .create(
      <VideoResult
        multiHighlight={{
          cut: [14, 239],
          highlights: [[18, 28]],
        }}
        onPlayClick={jest.fn()}
        sortBy="Containing"
        turn={0}
        turnContent="Good evening. The television and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency. The candidates need no introduction. The Republican candidate, Vice President Richard M. Nixon, and the Democratic candidate, Senator John F. Kennedy. According to rules set by the candidates themselves, each man shall make an opening statement of approximately eight minutes' duration and a closing statement of approximately three minutes' duration. In between the candidates will answer, or comment upon answers to questions put by a panel of correspondents. In this, the first discussion in a series of four - joint appearances, the subject matter has been agreed, will be restricted to internal or domestic American matters. And now for the first opening statement by Senator John F. Kennedy."
        videoFact={kennedyNixon}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("VideoResultPreview before and after", () => {
  const tree = renderer
    .create(
      <VideoResult
        multiHighlight={{
          cut: [0, 276],
          highlights: [[18, 28]],
        }}
        onPlayClick={jest.fn()}
        sortBy="BeforeAndAfter"
        turn={0}
        turnContent="Good evening. The television and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency. The candidates need no introduction. The Republican candidate, Vice President Richard M. Nixon, and the Democratic candidate, Senator John F. Kennedy. According to rules set by the candidates themselves, each man shall make an opening statement of approximately eight minutes' duration and a closing statement of approximately three minutes' duration. In between the candidates will answer, or comment upon answers to questions put by a panel of correspondents. In this, the first discussion in a series of four - joint appearances, the subject matter has been agreed, will be restricted to internal or domestic American matters. And now for the first opening statement by Senator John F. Kennedy."
        videoFact={kennedyNixon}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
