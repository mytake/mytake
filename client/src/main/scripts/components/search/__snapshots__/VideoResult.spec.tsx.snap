// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoResultPreview before and after 1`] = `
<div
  className="turn"
>
  <div
    className="turn__info"
  >
    <div
      className="turn__info-row"
    >
      <h3
        className="turn__speaker"
      >
         Smith
      </h3>
      <div
        className="dropdown dropdown--share"
      >
        <span
          className="dropdown__toggle dropdown__toggle--share"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
            />
            <polyline
              points="16 6 12 2 8 6"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="15"
            />
          </svg>
        </span>
        <div
          className="dropdown__dropdown dropdown__dropdown--hidden"
        >
          <SharePreview />
        </div>
      </div>
    </div>
    <div
      className="turn__info-row turn__info-row--short"
    >
      <span
        className="turn__time"
      >
        0:02 - 0:17
      </span>
    </div>
    <div
      className="turn__info-row"
    >
      <button
        className="turn__button turn__button--play"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={20}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="5 3 19 12 5 21 5 3"
          />
        </svg>
      </button>
      <button
        className="turn__button turn__button--bookmark"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <p
    className="turn__results"
  >
    Good evening. The 
    <strong>
      television
    </strong>
     and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency. The candidates need no introduction.
  </p>
</div>
`;

exports[`VideoResultPreview containing 1`] = `
<div
  className="turn"
>
  <div
    className="turn__info"
  >
    <div
      className="turn__info-row"
    >
      <h3
        className="turn__speaker"
      >
         Smith
      </h3>
      <div
        className="dropdown dropdown--share"
      >
        <span
          className="dropdown__toggle dropdown__toggle--share"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
            />
            <polyline
              points="16 6 12 2 8 6"
            />
            <line
              x1="12"
              x2="12"
              y1="2"
              y2="15"
            />
          </svg>
        </span>
        <div
          className="dropdown__dropdown dropdown__dropdown--hidden"
        >
          <SharePreview />
        </div>
      </div>
    </div>
    <div
      className="turn__info-row turn__info-row--short"
    >
      <span
        className="turn__time"
      >
        0:03 - 0:15
      </span>
    </div>
    <div
      className="turn__info-row"
    >
      <button
        className="turn__button turn__button--play"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={20}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="5 3 19 12 5 21 5 3"
          />
        </svg>
      </button>
      <button
        className="turn__button turn__button--bookmark"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <p
    className="turn__results"
  >
    The 
    <strong>
      television
    </strong>
     and radio stations of the United States and their affiliated stations are proud to provide facilities for a discussion of issues in the current political campaign by the two major candidates for the presidency.
  </p>
</div>
`;
